app/ui/omni-box.js

```diff
@@ -107,6 +109,34 @@ class OmniBox extends HTMLElement {
     this.forwardButton.addEventListener('click', () => {
       this.dispatchEvent(new CustomEvent('forward'))
     })
+
+    // middle mouse click paste&go
+    const win = new BrowserWindow()
+    this.input.addEventListener('paste-and-go', () => {
```

Comment:
`this.input` is an [HTML Input Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input) and doesn't have a built in method called "paste-and-go". This is why the code never gets triggered. When you middle click on the input it's pasting the event normally and your code isn't capturing that.

Instead you should be listening to `this.input.assEventListener("paste")` which is defined [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/paste_event)

The event has an `event.clipboardData` property and that's where you can see if the user pasted a URL or not.

Reply:
That makes sense.. thanks!